### ISSUE TITLE
TOTP field not detected on https://open.rocket.chat

### ISSUE THREAD

**Issue by foss-** [keepassxreboot/keepassxc-browser#1956]
Thanks for KeePassXC and KeePassXC-Browser 🙏

When logging into an account on https://open.rocket.chat/ with TOTP a modal overlay dialog asking for TOTP shows up.

## Current Behavior

However keepassxc-browser does not show the quick access icon to fill TOTP. It seems the field is not correctly detected or maybe there is information missing from the website?

## Possible Solution
Would be great to have the  quick access icon for TOTP.

## Steps to Reproduce
1. create account on https://open.rocket.chat
2. setup TOTP and logout
3. Login using email or username + password
4. keepassxc-browser not detecting TOTP field

## Debug info

KeePassXC KeePassXC - Version 2.8.0-snapshot, Build Type: Snapshot, Revision: 5fb26d6
KeePassXC-Browser 1.8.7
Operating system: macOS 13.5
Browser: Firefox 116.0.1

**Comment by varjolintu** [comment]
Can you provide the HTML code of that form and input field? Thanks.

**Comment by foss-** [comment]
`<input class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-text rcx-input-box" type="text" id="xxxxxxxxxxx" placeholder="Enter authentication code" size="1" value="">`

Apparently RC is making this unnecessarily hard. Filed https://github.com/RocketChat/Rocket.Chat/issues/30025
Your call whether to keep this issue here open as reference or not.

**Comment by varjolintu** [comment]
Sorry, I edited the comment a few times. You can quote the whole text in the RocketChat issue.

**Comment by varjolintu** [comment]
Btw, have you tried to use Custom Login Fields for the field?

**Comment by foss-** [comment]
No, is there documentation about that? A bit thin on time currently tough.

**Comment by varjolintu** [comment]
@foss- Seems.. there's not (gotta fix that). Basically when you are on the login screen, open the extension popup from the icon and click the yellow Custom Login Fields icon. Select username/password fields and confirm. Then when you are on the 2FA page, open the selector again and save a TOTP field.

**Comment by foss-** [comment]
Quite the capability. Don't think 1Password has such functionality? Creating the custom field works as expected and the bubble icon for TOTP now shows for the TOTP dialog.

**Comment by varjolintu** [comment]
> Quite the capability. Don't think 1Password has such functionality? Creating the custom field works as expected and the bubble icon for TOTP now shows for the TOTP dialog.

Not sure about other password managers. As far as I know many has a possibility to put an extra element ID etc. to the entry properties, and the extension tries to identify that. This kind of (a very basic one) functionality was already in the original passifox/chromeipass, but our version is totally redesigned for the purpose, including support for iframes etc.

**Comment by varjolintu** [comment]
Closing this because there's no easy solution to detect the field as 2FA and Custom Login Fields solves the issue.
